### PR TITLE
Feat/588 token price second design iteration

### PR DESCRIPTION
**Pull request by joshuanatanielnm**
#588 

## Description

Add 1M Change vs ETH/BTC 50/50 for token price second design iteration

## Changes

List any technical changes.

- [x] [feat: create hooks for eth and btc interval gain](https://github.com/strangelove-ventures/sommelier/commit/29f2866a7b97428ecd77e1b0c745c8b249a0ace6)
- [x] [feat: create actions to get assets interval gain](https://github.com/strangelove-ventures/sommelier/commit/b4c003e2694b317a44dc97cae7a1dd4814738522)
- [x] [feat: create utils to get gain pct data](https://github.com/strangelove-ventures/sommelier/commit/13bebdec78bf2ab0f08a6e56b94fe6ea3710f8f2)
- [x] [feat: add get today and previous month](https://github.com/strangelove-ventures/sommelier/commit/56d936fc39f0cd7a541b4c6b17e0b600d418f73e)
- [x] [feat: create query to get single cellar value](https://github.com/strangelove-ventures/sommelier/commit/32c5b0ed81f2ccabc78d7746dd22eb22255237c9) 
- [x] [feat: getPreviousMonth calculation](https://github.com/strangelove-ventures/sommelier/commit/8f1c7ceaa925d148a35a0731a7df8f59ca4dc225) 
- [x] [feat: add intervalGainPct in about cellar](https://github.com/strangelove-ventures/sommelier/commit/e34c2a8d2e801b27f03df4cdcdad403fd96fad2e)
- [x] [feat: add intervalGainPct in page cellar](https://github.com/strangelove-ventures/sommelier/commit/ec278038dc25c4a05eb9ccbd8296926d23eeb2d7)
- [x] [feat: add intervalGainPct in hero section](https://github.com/strangelove-ventures/sommelier/commit/e82bca6ad3fb22c5616d85c5c16e1a399dcc825f)

## Screenshots (if appropriate):
image
image
image

## Links
https://www.figma.com/file/7IAQbLyzapnrow9sIcO77P/App?node-id=2195:101054

### CODE REVIEW

File: src/components/CellarStatsAutomated.tsx [hunk 1]
```diff
@@ -100,6 +107,29 @@ export const CellarStatsAutomated: VFC<CellarStatsAutomatedProps> = ({
           </Tooltip>
         </Box>
       </VStack>
+      <VStack spacing={1} align="center" w="7rem">
+        {monthChangeValue}
+        <Box
+          onMouseEnter={debounce(() => {
+            analytics.track("user.tooltip-opened-daily-change")
```
Comment (codingki) [inline src/components/CellarStatsAutomated.tsx:114]: tooltip-opened-monthly-change

File: src/data/hooks/useIntervalGainPct.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      Boolean(ethIntervalGain.data) &&
+      Boolean(btcIntervalGain.data)
+  )
+
+  const query = useQuery(
+    [
+      "USE_INTERVAL_GAIN_PCT",
+      config.id,
+      todayDatas,
+      previousMonthDatas,
```
Comment (codingki) [inline src/data/hooks/useIntervalGainPct.ts:48]: previousMonthDatas[0].shareValue

**Review by codingki — CHANGES_REQUESTED** [review]

File: src/utils/calculateTime.ts [hunk 3]
```diff
@@ -15,3 +24,16 @@ export const getPreviousWeek = (): number => {
...
+
+export const getPreviousMonth = (): number => {
+  const dayInSecs = 24 * 60 * 60
+  const now = new Date()
+  const previousMonth =
+    Math.floor(
+      (now.getTime() - dayInSecs * 30 * 1000) / 1000 / dayInSecs
+    ) *
+      dayInSecs +
+    dayInSecs * 3
```
Comment (codingki) [inline src/utils/calculateTime.ts:36]: this is weird, I don't know about the calculation, but by reading this seems make no sense it's a previous month cc @xanderjl @sandbochs

**Review by xanderjl — CHANGES_REQUESTED** [review]
All of the logic is very close! Just a few best practice things I'd change and I'd say this is good to go. Great work @joshuanatanielnm 😄

File: src/components/HeroStrategy/hero-right.tsx [hunk 4]
```diff
@@ -59,17 +61,18 @@ export const HeroStrategyRight: VFC<HeroStrategyRightProps> = ({
       <HStack
         pt={4}
         justifyContent="space-around"
+        alignItems="start"
         divider={<StackDivider borderColor="purple.dark" />}
       >
-        <VStack>
+        <VStack w="30%">
```
[the remark this answers is not in the document]
Comment (joshuanatanielnm) [reply src/components/HeroStrategy/hero-right.tsx:85]: Thanks Xan that solve the issue

File: src/components/PercentageText.tsx [hunk 5]
```diff
@@ -26,7 +28,18 @@ export const PercentageText: VFC<PercentageTextProps> = ({
...
+            <FaArrowDown />
+          ) : (
+            <Heading size={headingSize}>-</Heading>
+          )
+        ) : arrow ? (
+          <FaArrowUp />
+        ) : (
+          <Heading size={headingSize}>+</Heading>
+        ))}
+
```
[the remark this answers is not in the document]
Comment (joshuanatanielnm) [reply src/components/PercentageText.tsx:42]: Thanks Xan 🚀

**Review by codingki — APPROVED** [review]

**Pull request merged**